library/ssl_tls.c

```diff
@@ -3089,8 +3122,8 @@ void mbedtls_ssl_handshake_free( mbedtls_ssl_context *ssl )
     mbedtls_ssl_buffering_free( ssl );
 #endif /* MBEDTLS_SSL_PROTO_DTLS */
 
-#if defined(MBEDTLS_ECDH_C) &&                  \
-    defined(MBEDTLS_USE_PSA_CRYPTO)
+#if defined(MBEDTLS_ECDH_C) && \
+    ( defined(MBEDTLS_USE_PSA_CRYPTO) || defined(MBEDTLS_SSL_PROTO_TLS1_3) )
```

Comment:
Both `MBEDTLS_USE_PSA_CRYTO` and `MBEDTLS_SSL_PROTO_TLS1_3` depend on `MBEDTLS_PSA_CRYPTO_C`
```suggestion
#if defined(MBEDTLS_ECDH_C) && defined(MBEDTLS_PSA_CRYPTO_C) 
```

Reply:
Yes that would be another possibility but the current version reflects better the current status of the code base regarding usage of PSA in TLS I'd say. Thus not changed.